File: 238-product-of-array-except-self/product-of-array-except-self.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int len = nums.length;
        int res[] = new int[len];
        int curr = 1;
        Arrays.fill(res, 1);
        for (int i = 0; i < len; i++) {
            res[i] *= curr;
            curr *= nums[i];
        }
        curr = 1;
        for (int i = len - 1; i > -1; i--) {
            res[i] *= curr;
            curr *= nums[i];
        }
        return res;
    }
}

File: 238-product-of-array-except-self/product-of-array-except-self.ts
function productExceptSelf(nums: number[]): number[] {
    const len: number = nums.length;
    let res: number[] = new Array(len).fill(1);
    let curr: number = 1;
    for (let i = 0; i < len; i += 1) {
        res[i] *= curr;
        curr *= nums[i];
    }
    curr = 1;
    for (let i = len - 1; i > -2; i -= 1) {
        res[i] *= curr;
        curr *= nums[i];
    }
    return res;
};